src/main/lrsql/system/webserver.clj

```diff
@@ -71,7 +74,12 @@
      ::i/enable-statement-html enable-stmt-html
      ::http/allowed-origins
      {:creds           true
-      :allowed-origins (constantly true)}
+      :allowed-origins (fn [origin]
+                         (or allow-all-origins
+                             (some #(= origin %)
+                                   (or allowed-origins
+                                       [(format "http://%s:%s" http-host http-port)
+                                        (format "https://%s:%s" http-host ssl-port)]))))}
```

Comment:
This should be extracted into a helper function (and also maybe extract the array of formatted IRIs into its own `let` binding)

Reply:
Or in general just make these lines of code a bit less horrifying.

Reply:
done